src/editors/framework/content-analysis.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Yoast\WP\SEO\Editors\Framework;
+
+use Yoast\WP\SEO\Helpers\Options_Helper;
+
+/**
+ * This class describes the Content analysis feature.
+ */
+class Content_Analysis implements Analysis_Feature_Interface {
+
+	/**
+	 * The options helper.
+	 *
+	 * @var Options_Helper
+	 */
+	private $options_helper;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param Options_Helper $options_helper The options helper.
+	 */
+	public function __construct( Options_Helper $options_helper ) {
+		$this->options_helper = $options_helper;
+	}
+
+	/**
+	 * Whether this analysis is enabled.
+	 *
+	 * @return bool Whether this analysis is enabled.
+	 */
+	public function is_enabled(): bool {
+		return $this->is_globally_enabled() && $this->is_user_enabled();
+	}
+
+	/**
+	 * Whether this analysis is enabled by the user.
+	 *
+	 * @return bool Whether this analysis is enabled by the user.
+	 */
+	private function is_user_enabled(): bool {
+		return ! \get_the_author_meta( 'wpseo_content_analysis_disable', \get_current_user_id() );
```

Comment:
I think expanding the interface right now is a bit overkill, if we ever need to support those cases it would be very easy to expand the interface or add an additional method for this specific case.

I will change the author_meta call. When looking into it, it seems to be made to get data for the specific author of a post, which we overwrite with the current_user_id...